src/ray/common/asio/asio_util.h

```diff
@@ -15,22 +15,28 @@
 #pragma once
 
 #include <boost/asio.hpp>
+#include <chrono>
 
-inline std::shared_ptr<boost::asio::deadline_timer> execute_after_us(
+template <typename Duration>
+std::shared_ptr<boost::asio::deadline_timer> execute_after(
     instrumented_io_context &io_context,
     std::function<void()> fn,
-    int64_t delay_microseconds) {
+    Duration delay_duration) {
   auto timer = std::make_shared<boost::asio::deadline_timer>(io_context);
-  timer->expires_from_now(boost::posix_time::microseconds(delay_microseconds));
+  if constexpr (std::is_integral_v<Duration>) {
```

Comment:
Hmm wonder if it is better just changing all downstream callers code instead of handling it this way? There seems to be only 5~10 changes needed.

Reply:
Or alternatively you can just write a docstring the default is ms...